frontend/src/libs/packages/storage/storage.package.ts

```diff
@@ -10,26 +10,62 @@ class Storage implements IStorage {
     this.store = store;
   }
 
-  public set(key: ValueOf<typeof StorageKey>, value: string): Promise<void> {
-    this.store.setItem(key as string, value);
+  public set<T>(key: keyof typeof StorageKey, value: T): Promise<void> {
+    try {
+      const serializedValue = JSON.stringify(value);
+      this.store.setItem(key as string, serializedValue);
 
-    return Promise.resolve();
+      return Promise.resolve();
+    } catch (error) {
+      return Promise.reject(
+        new ApplicationError({
+          message: `Error setting value in storage '${key}'`,
+          cause: error,
+        }),
+      );
+    }
   }
 
-  public get<R = string>(key: ValueOf<typeof StorageKey>): Promise<R | null> {
-    return Promise.resolve(this.store.getItem(key as string) as R);
+  public get<T>(key: keyof typeof StorageKey): Promise<T | null> {
+    try {
+      const value = this.store.getItem(key as string);
+
+      return value === null
+        ? Promise.resolve(null)
+        : Promise.resolve(JSON.parse(value) as T);
+    } catch (error) {
+      return Promise.reject(
+        new ApplicationError({
+          message: `Error getting value from storage '${key}'`,
+          cause: error,
+        }),
+      );
+    }
   }
 
-  public drop(key: ValueOf<typeof StorageKey>): Promise<void> {
-    this.store.removeItem(key as string);
+  public drop(key: keyof typeof StorageKey): Promise<void> {
```

Comment:
could you declare type above Storage class called StorageKeyValue = keyof typeof StorageKey and use it here 

key: StorageKeyValue